File: java/sdk/src/main/java/cn/huidu/device/sdk/data/program/area/AreaNode.java
package cn.huidu.device.sdk.data.program.area;

import com.alibaba.fastjson.JSONArray;
import com.alibaba.fastjson.JSONObject;

import cn.huidu.device.sdk.data.program.area.content.ContentNode;
import cn.huidu.device.sdk.data.program.area.content.DialClockNode;
import cn.huidu.device.sdk.data.program.area.content.DigitalClockNode;
import cn.huidu.device.sdk.data.program.area.content.DynamicNode;
import cn.huidu.device.sdk.data.program.area.content.ImageNode;
import cn.huidu.device.sdk.data.program.area.content.TextNode;
import cn.huidu.device.sdk.data.program.area.content.VideoNode;
import cn.huidu.device.sdk.data.program.other.Border;

public class AreaNode extends ContentNode {
    private int x;
    private int y;
    private int width;
    private int height;
    private ContentNode[] item;
    private Border border;

    public AreaNode() {
        super();
    }

    public AreaNode(JSONObject jsonObj) {
        this(jsonObj.toJavaObject(AreaNode.class));
        // 重新初始化内容插件
        JSONArray jsons = jsonObj.getJSONArray("item");
        if (jsons.size() == item.length) {
            for (int i = 0; i < item.length; i++) {
                if (item[i].getType() == null) {
                    continue;
                }

                switch (item[i].getType()) {
                    case "text": {
                        item[i] = jsons.getJSONObject(i).toJavaObject(TextNode.class);
                    }
                        break;
                    case "dynamic": {
                        item[i] = jsons.getJSONObject(i).toJavaObject(DynamicNode.class);
                    }
                        break;
                    case "image": {
                        item[i] = jsons.getJSONObject(i).toJavaObject(ImageNode.class);
                    }
                        break;
                    case "video": {
                        item[i] = jsons.getJSONObject(i).toJavaObject(VideoNode.class);
                    }
                        break;
                    case "digitalClock": {
                        item[i] = jsons.getJSONObject(i).toJavaObject(DigitalClockNode.class);
                    }
                        break;
                    case "dialClock": {
                        item[i] = jsons.getJSONObject(i).toJavaObject(DialClockNode.class);
                    }
                        break;
                    default:
                        break;
                }
            }
        }
    }

    public AreaNode(ContentNode contents) {
        this.item = new ContentNode[] { contents };
    }

    public AreaNode(ContentNode[] contents) {
        this.item = contents;
    }

    protected AreaNode(AreaNode other) {
        super(other);
        this.x = other.x;
        this.y = other.y;
        this.width = other.width;
        this.height = other.height;
        this.item = other.item;
        this.border = other.border;
    }

    public int getX() {
        return x;
    }

    public void setX(int x) {
        this.x = x;
    }

    public int getY() {
        return y;
    }

    public void setY(int y) {
        this.y = y;
    }

    public int getWidth() {
        return width;
    }

    public void setWidth(int width) {
        this.width = width;
    }

    public int getHeight() {
        return height;
    }

    public void setHeight(int height) {
        this.height = height;
    }

    public ContentNode[] getItem() {
        return item;
    }

    public void setItem(ContentNode[] item) {
        this.item = item;
    }

    public Border getBorder() {
        return border;
    }

    public void setBorder(Border border) {
        this.border = border;
    }
}

File: python/sdk/data/area/AreaNode.py
from typing import List, Optional, Dict, Any, Type
from sdk.data.area.ContentNode import ContentNode
from sdk.data.other.Border import Border
from sdk.data.area import (
    TextNode, DynamicNode, ImageNode, VideoNode, DigitalClockNode, DialClockNode
)


class AreaNode(ContentNode):
    CONTENT_TYPE_MAP: Dict[str, Type[ContentNode]] = {
        "text": TextNode,
        "dynamic": DynamicNode,
        "image": ImageNode,
        "video": VideoNode,
        "digitalClock": DigitalClockNode,
        "dialClock": DialClockNode
    }

    def __init__(self, *args):
        super().__init__()
        self.x: int = 0
        self.y: int = 0
        self.width: int = 0
        self.height: int = 0
        self.item: List[ContentNode] = []
        self.border: Optional[Border] = None

        if len(args) == 1:
            arg = args[0]
            if isinstance(arg, ContentNode):
                # Single content node constructor
                self.item = [arg]
            elif isinstance(arg, list) and all(isinstance(i, ContentNode) for i in arg):
                # List of content nodes
                self.item = arg
            elif isinstance(arg, AreaNode):
                # Copy constructor
                self._copy_from(arg)
            elif isinstance(arg, dict):
                # JSON dictionary constructor
                self._from_dict(arg)

    def _copy_from(self, other: 'AreaNode'):
        super()._copy_from(other)
        self.x = other.x
        self.y = other.y
        self.width = other.width
        self.height = other.height
        self.item = other.item.copy()  # Shallow copy
        self.border = Border(other.border) if other.border else None

    def _from_dict(self, data: Dict[str, Any]):
        if 'uuid' in data:
            self.id = data['uuid']
        if 'name' in data:
            self.name = data['name']
        if 'type' in data:
            self.type = data['type']
        if 'bindType' in data:
            self.bindType = data['bindType']

        self.x = data.get('x', 0)
        self.y = data.get('y', 0)
        self.width = data.get('width', 0)
        self.height = data.get('height', 0)

        if 'border' in data:
            self.border = Border(**data['border']) if data['border'] else None

        self.item = []
        for item_data in data.get('item', []):
            content_type = item_data.get('type')
            if content_type in self.CONTENT_TYPE_MAP:
                node_class = self.CONTENT_TYPE_MAP[content_type]
                self.item.append(node_class(item_data))

    def get_x(self) -> int:
        return self.x

    def set_x(self, x: int):
        self.x = x

    def get_y(self) -> int:
        return self.y

    def set_y(self, y: int):
        self.y = y

    def get_width(self) -> int:
        return self.width

    def set_width(self, width: int):
        self.width = width

    def get_height(self) -> int:
        return self.height

    def set_height(self, height: int):
        self.height = height

    def get_item(self) -> List[ContentNode]:
        return self.item

    def set_item(self, items: List[ContentNode]):
        self.item = items

    def get_border(self) -> Optional[Border]:
        return self.border

    def set_border(self, border: Border):
        self.border = border

    def to_dict(self) -> Dict[str, Any]:
        data = super().to_dict()
        data.update({
            "x": self.x,
            "y": self.y,
            "width": self.width,
            "height": self.height,
            "item": [item.to_dict() for item in self.item],
            "border": self.border.to_dict() if self.border else None
        })
        return data

    def __repr__(self) -> str:
        return (f"AreaNode(uuid={self.id}, name='{self.name}', "
                f"x={self.x}, y={self.y}, width={self.width}, height={self.height}, "
                f"items={len(self.item)}, border={self.border is not None})")